### PR TITLE
poppler 24.09.0: rebuild with libtiff 4.5.1 and licms2 2.16

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** defaults

### Links

- [PKG-5339](https://anaconda.atlassian.net/browse/PKG-6506)
- [Upstream repository](https://gitlab.freedesktop.org/poppler/poppler)
- [Upstream changelog/diff](https://gitlab.freedesktop.org/poppler/poppler/-/blob/poppler-24.09.0/NEWS?ref_type=tags)
- Requirements: 

### Explanation of changes:

- Bump the build number
- Pin libtiff to 4.5.1 and licms2 to 2.16

### Notes

- https://github.com/AnacondaRecipes/libtiff-feedstock/pull/20
- https://github.com/AnacondaRecipes/lcms2-feedstock/pull/1

[PKG-5339]: https://anaconda.atlassian.net/browse/PKG-5339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by bwingconda — APPROVED** [review]

**Review by SC426 — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -36,6 +36,7 @@ build:
 
 requirements:
   build:
+    # gobject-introspection 1.* requires setuptools at runtime, but
     # setuptools >= 65.0.0 is lacking the msvccompiler module in distutils.
     # https://github.com/pypa/setuptools/pull/3505
     - setuptools <65
```
Comment (scotthain) [inline recipe/meta.yaml:42]: Can we patch this from `msvccompiler` to `_msvccompiler`? It looks like in modern setuptools it exists but it obfuscated. I'm not sure if it's worth the effort though? https://github.com/pypa/setuptools/blob/main/setuptools/_distutils/_msvccompiler.py

**Review by scotthain — APPROVED** [review]
One question but nothing blocking.

File: recipe/meta.yaml (hunk 1 above)
Comment (scotthain) [reply recipe/meta.yaml:42]: Oh It's a build time dep - nevermind! Need more caffiene!

**Comment by skupr-anaconda** [comment]
If I use libtiff {{ libtiff }} in host on win then:
```
ExplainedDependencyNeedsBuildingError: Unsatisfiable dependencies for platform win-64: {MatchSpec("lcms2==2.16=h595ce26_0"), MatchSpec("libtiff[version='>=4.5.1,<5.0a0']")}

Encountered problems while solving:

  - package lcms2-2.16-h595ce26_0 requires libtiff >=4.5.1,<5.0a0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible

\u251c\u2500 lcms2 2.16.*  is installable and it requires
\u2502  \u2514\u2500 libtiff >=4.5.1,<5.0a0 , which can be installed;
\u2514\u2500 libtiff 4.5.0.*  is not installable because it conflicts with any installable versions previously reported.
```
See also https://github.com/AnacondaRecipes/lcms2-feedstock/pull/1#discussion_r1879810767

I rewrote it to:
```
    # libtiff 4.5.1 fixes a missing libtiff.lib on Windows
    - libtiff 4.5.1          # [win]
    - libtiff {{ libtiff }}  # [not win]
```

**Pull request merged**